provider/aci.go

```diff
@@ -1639,38 +1642,45 @@ func getProbe(probe *v1.Probe, ports []v1.ContainerPort) (*aci.ContainerProbe, e
 }
 
 func (p *ACIProvider) getAzureFileCSI(volume v1.Volume, namespace string) (*aci.Volume, error) {
-	azureSource := &aci.AzureFileVolume{
-		ReadOnly: *volume.CSI.ReadOnly,
+	azureSource := &aci.AzureFileVolume{}
+	var secretName, shareName string
+	if volume.CSI.VolumeAttributes != nil && len(volume.CSI.VolumeAttributes) != 0 {
+		for k, v := range volume.CSI.VolumeAttributes {
+			switch strings.ToLower(k) {
+			case azureFileSecretName:
+				secretName = v
+			case azureFileShareName:
+				shareName = v
+			}
+		}
+	} else {
+		return nil, fmt.Errorf("secret volume attribute for AzureFile CSI driver %s cannot be empty or nil", volume.Name)
 	}
-	if volume.CSI.NodePublishSecretRef != nil && volume.CSI.NodePublishSecretRef.Name != "" {
 
-		// Set the storage account name and key
-		secret, err := p.resourceManager.GetSecret(volume.CSI.NodePublishSecretRef.Name, namespace)
-		if err != nil {
-			return nil, errors.Wrap(err, fmt.Sprintf("The secret %s for AzureFile CSI driver %s is not found", volume.CSI.NodePublishSecretRef.Name, volume.Name))
-		}
+	if shareName == "" {
+		return nil, fmt.Errorf("secret share name for AzureFile CSI driver %s cannot be empty or nil", volume.Name)
+	}
 
-		if secret == nil {
-			return nil, fmt.Errorf("Getting secret for AzureFile CSI driver %s returned an empty secret", volume.Name)
-		}
+	// Set shareName
+	azureSource.ShareName = shareName
 
-		azureSource.StorageAccountName = string(secret.Data["azurestorageaccountname"])
-		azureSource.StorageAccountKey = string(secret.Data["azurestorageaccountkey"])
+	secret, err := p.resourceManager.GetSecret(secretName, namespace)
 
-		// Set shareName
-		if volume.CSI.VolumeAttributes != nil {
-			if shareName, ok := volume.CSI.VolumeAttributes[azureFileShareName]; ok {
-				azureSource.ShareName = shareName
-			}
-		}
-		volume := aci.Volume{
-			Name:      volume.Name,
-			AzureFile: azureSource,
-		}
-		return &volume, nil
-	} else {
-		return nil, fmt.Errorf("NodePublishSecretRef for AzureFile CSI driver %s cannot be empty or nil", volume.Name)
+	if err != nil {
+		return nil, errors.Wrap(err, fmt.Sprintf("the secret %s for AzureFile CSI driver %s is not found", secretName, volume.Name))
 	}
+
+	if secret == nil {
+		return nil, fmt.Errorf("getting secret for AzureFile CSI driver %s returned an empty secret", volume.Name)
+	}
+
+	// Set the storage account name and key
+	azureSource.StorageAccountName = strings.TrimSpace(string(secret.Data[azureFileStorageAccountName]))
```

Comment:
what if the secret doesn't contain that azureFileStorageAccountName key?

Reply:
Good call. Added this corner case along with the unit tests

Reply:
@ryanzhang-oss  according to Andy's comment [here](https://github.com/virtual-kubelet/azure-aci/pull/199#discussion_r870896485), no need for this check.